src/dune_engine/load_rules.ml

```diff
@@ -501,14 +501,15 @@ end = struct
 
   module Normal = struct
     type t =
-      { build_dir_only_sub_dirs : Subdir_set.t
+      { build_dir_only_sub_dirs : Build_config.Rules.Build_only_sub_dirs.t
```

Comment:
In `make_rules_gen_result` we verify that both `rules` and `directory_targets` only contain valid keys. We should similar validation for this map as well.

Reply:
did that in [72bea81](https://github.com/ocaml/dune/pull/7207/commits/72bea812588daf8a2c34c1a5ee2d566cfaa3c5ab)